program/src/instruction.rs

```diff
@@ -1,184 +1,420 @@
-//! Program instructions
-
+use shank::{ShankContext, ShankInstruction, ShankType};
 use solana_program::clock::UnixTimestamp;
 
-/// Instructions supported by the staking program
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub enum StakeInstruction {
-    /// Creates Stake config account which controls staking parameters
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[]` Slash authority, may be a PDA for a governance program
-    /// 2. `[]` Config authority, may be a PDA for a governance program
-    /// 3. `[]` Stake Token Mint
-    /// 4. `[]` Stake Token Vault, to hold all staked tokens.
-    ///   Must be empty and owned by PDA with seeds `['token-owner', stake_config]`
+/// Enum defining all instructions in the Stake program.
+#[rustfmt::skip]
+#[derive(Clone, Debug, ShankContext, ShankInstruction)]
+pub enum Instruction {
+    /// Creates Stake config account which controls staking parameters.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        name = "slash_authority",
+        desc = "Slash authority"
+    )]
+    #[account(
+        2,
+        name = "config_authority",
+        desc = "Config authority"
+    )]
+    #[account(
+        3,
+        name = "mint",
+        desc = "Stake token mint"
+    )]
+    #[account(
+        4,
+        name = "vault",
+        desc = "Stake token vault"
+    )]
     InitializeConfig {
-        cooldown_time: UnixTimestamp,
+        cooldown_time_seconds: UnixTimestamp,
```

Comment:
Btw, if `UnixTimestamp` is annoying, feel free to just use an `i64`

Reply:
It would be easier to use `i64`. I also just realised that there is a bit of an inconsistency: we have `cooldown_time_seconds: UnixTimestamp` on the `InitializeConfig` and `CooldownTimeSeconds(u64)` on the `UpdateConfig`. Should both be `i64` in this case?

Reply:
Ah right... it doesn't make sense to have a negative cooldown time, so how about we just move everything to u64s?